Fixed some typos (#2066)
dipy/dipy: doc/devel/coding_style_guideline.rst

The main principles behind DIPY_ development are:

* **Robustness**: the results of a piece of code must be verified

>   systematically, and hence stability and robustness of the code must be ensured, reducing code redundancies.

Reviewer: ```suggestion
  systematically, and hence stability and robustness of the code must be 
  ensured, reducing code redundancies.
```

Reviewer: Was this just an elimination of the new line? I wouldn't stretch it out that long.

Author: Oh! Thanks for the suggestion

* **Readability**: the code is written and read by humans, and it is read
  much more frequently than it is written.
* **Consistency**: following these guidelines will ease reading the code,
  and will make it less error-prone.
* **Documentation**: document the code. Documentation is essential as it is

>   one of the key points for the adoption of DIPY as the toolkit of choice in diffusion by the scientific community. Documenting helps to clarify certain choices, helps to avoid obscure places, and is a way to allow other members *decode* it with less effort.

Reviewer: I would break this up a bit and streamline a tiny bit, like so:

```suggestion
  one of the key points for the adoption of DIPY as the toolkit of choice in 
  diffusion by the scientific community. Documenting helps to clarify certain 
  choices, helps to avoid obscure places, allowing other members 
  to *decode* it with less effort.
```

Author: Okay sure will revert that

`PEP8 <https://www.python.org/dev/peps/pep-0008/>`_ style when writing
`Cython <https://cython.org/>` code.

> The cython-specific syntax should follow these additional rules:

Reviewer: I would revert this one:

```suggestion
Cython-specific syntax should follow these additional rules:
```

Author: Yea sure

* The classes, objects, and any other construct referenced from the code
  should be written with inverted commas, such as in *In DIPY, we use an
  object called ``GradientTable`` which holds all the acquisition specific
  parameters, e.g. b-values, b-vectors, timings, and others.*
* Cite the relevant papers. Use the *[NameYear]* convention for

>   cross-referencing them, such as in [Garyfallidis2014]_, and put them under the :ref:`references` section.

Reviewer: ```suggestion
  cross-referencing them, such as in [Garyfallidis2014]_, and put them 
  under the :ref:`references` section.
```

* Cross-reference related examples and files. Use the
  ``.. _specific_filename:`` convention to label a file at the top of it.
  Thus, other pages will be able to reference the file using the standard
  Sphinx syntax ``:ref:`specific_filename```.
* Use an all-caps scheme for acronyms, and capitalize the first letters of
  the long names, such as in *Constrained Spherical Deconvolution (CSD)*,

>   except in those cases where the most common convention has been to use lowercase, such as in *superior longitudinal fasciculus (SLF)*.

Reviewer: ```suggestion
  except in those cases where the most common convention has been to use 
  lowercase, such as in *superior longitudinal fasciculus (SLF)*.
```

Author: I will do the following changes

* As customary in Python, use lowercase and separate words with underscores
  for filenames, labels for references, etc.
* When including figures, use the regular font for captions (i.e. do not use
  bold faces) unless otherwise required for a specific text part (e.g. a
  DIPY object, etc.).
* When referring to relative paths, use the backquote inline markup

>   the convention, such as in ``doc/devel``. Do not add the

Reviewer: I'd rever this suggestion

```suggestion
  convention, such as in ``doc/devel``. Do not add the
```

Author: Yea will do the reversion
